### PR TITLE
add the get release-related-files endpoints

### PR DESCRIPTION
**Pull request by dubdabasoduba**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 		<redis.lettuce.version>5.2.2.RELEASE</redis.lettuce.version>
 		<opensrp.updatePolicy>always</opensrp.updatePolicy>
 		<nexus-staging-maven-plugin.version>1.5.1</nexus-staging-maven-plugin.version>
-		<opensrp.core.version>2.4.3-SNAPSHOT</opensrp.core.version>
+		<opensrp.core.version>2.4.4-SNAPSHOT</opensrp.core.version>
```
Comment (ekigamba) [inline pom.xml]: Change to 2.1.26

File: src/main/java/org/opensrp/web/rest/ClientFormResource.java [hunk 2]
```diff
@@ -141,6 +151,42 @@ private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier)
...
+        Manifest manifest = manifestService.getManifest(releaseIdentifier);
+        if (manifest != null && StringUtils.isNotBlank(manifest.getJson())) {
+            JSONObject json = new JSONObject(manifest.getJson());
+            if (json.has(FORM_IDENTIFIERS)) {
+                JSONArray fileIdentifiers = json.getJSONArray(FORM_IDENTIFIERS);
+                if (fileIdentifiers != null && fileIdentifiers.length() > 0) {
+                    for (int i = 0; i < fileIdentifiers.length(); i++) {
+                        String fileIdentifier = fileIdentifiers.getString(i);
+                        ClientFormMetadata clientFormMetadata = getMostRecentVersion(fileIdentifier);
+                        clientFormMetadataList.add(clientFormMetadata);
```
Comment (ekigamba) [inline src/main/java/org/opensrp/web/rest/ClientFormResource.java:184]: Check for null

File: src/main/java/org/opensrp/web/rest/ClientFormResource.java [hunk 3]
```diff
@@ -141,6 +151,42 @@ private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier)
...
+                } else {
+                    return new ResponseEntity<>("This manifest does not have any files related to it",
+                                HttpStatus.NO_CONTENT);
+                }
+            } else {
+                return new ResponseEntity<>("This manifest does not have any files related to it",
+                       HttpStatus.NO_CONTENT);
+            }
+        } else {
+            return new ResponseEntity<>("No manifest matching the given identifier was found",
```
Comment (ekigamba) [inline src/main/java/org/opensrp/web/rest/ClientFormResource.java:195]: A bit misleading, just add the ` OR manifest does not have any files related to it`

File: pom.xml (hunk 1 above)
Comment (dubdabasoduba) [reply pom.xml]: 2.1.25 is not tagged yet we can tag this there too

**Pull request merged**